// addons-scanner-utils/dist/functions/error.js
"use strict";
/* eslint-disable max-classes-per-file */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AMOError = exports.createAppError = exports.AppError = void 0;
/**
 * Error with an HTTP status code and optional extra information intended to be
 * used by the error handler defined in `src/functions/index.ts`.
 *
 * @property extraInfo - Additional error details
 * @property status - HTTP status code
 */
class AppError extends Error {
    constructor({ message, extraInfo, status = 500 }) {
        super(message);
        this.name = 'AppError';
        this.status = status;
        this.extraInfo = extraInfo;
    }
}
exports.AppError = AppError;
/**
 * Create an app error object with status code and extra information.
 */
const createAppError = (params) => {
    return new AppError(params);
};
exports.createAppError = createAppError;
/**
 * Error object used when interactions with the AMO API led to errors.
 */
class AMOError extends Error {
    constructor(message) {
        super(message);
        this.name = 'AMOError';
    }
}
exports.AMOError = AMOError;
